src/main/java/org/opensearch/ad/transport/ADJobRunnerTransportAction.java

```diff
@@ -14,66 +14,65 @@
 
     private static final Logger LOG = LogManager.getLogger(ADJobRunnerTransportAction.class);
 
     private SDKRestClient client;
+    private final SDKNamedXContentRegistry xContentRegistry;
+    private final Settings settings;
 
-    private AnomalyDetectorJob scheduledJobParameter;
-
-    private final SDKNamedXContentRegistry namedXContentRegistry;
-
+    @Inject
     protected ADJobRunnerTransportAction(
-        TransportService transportService,
+        ExtensionsRunner extensionsRunner,
         ActionFilters actionFilters,
-        SDKRestClient client,
-        ExtensionsRunner extensionsRunner
+        TaskManager taskManager,
+        SDKNamedXContentRegistry xContentRegistry,
+        SDKRestClient client
     ) {
-        super(ADJobRunnerAction.NAME, transportService, actionFilters, ExtensionActionRequest::new);
+        super(ADJobRunnerAction.NAME, actionFilters, taskManager);
         this.client = client;
-        this.namedXContentRegistry = extensionsRunner.getNamedXContentRegistry();
+        this.xContentRegistry = xContentRegistry;
+        this.settings = extensionsRunner.getEnvironmentSettings();
     }
 
     @Override
-    protected void doExecute(Task task, ExtensionActionRequest request, ActionListener<ExtensionActionResponse> actionListener) {
+    protected void doExecute(Task task, JobRunnerRequest request, ActionListener<JobRunnerResponse> actionListener) {
         String errorMessage = "Failed to run the Job";
-        ActionListener<ExtensionActionResponse> listener = wrapRestActionListener(actionListener, errorMessage);
-        JobRunnerRequest jobRunnerRequest;
+        ActionListener<JobRunnerResponse> listener = wrapRestActionListener(actionListener, errorMessage);
         try {
-            jobRunnerRequest = new JobRunnerRequest(request.getRequestBytes());
-            CompletableFuture<AnomalyDetectorJob> inProgressFuture = new CompletableFuture<>();
-            String jobParameterDocumentId = jobRunnerRequest.getJobParameterDocumentId();
+            JobExecutionContext jobExecutionContext = request.getJobExecutionContext();
+            String jobParameterDocumentId = jobExecutionContext.getJobId();
             if (jobParameterDocumentId == null || jobParameterDocumentId.isEmpty()) {
                 listener.onFailure(new IllegalArgumentException("jobParameterDocumentId cannot be empty or null"));
             } else {
+                CompletableFuture<AnomalyDetectorJob> inProgressFuture = new CompletableFuture<>();
                 findById(jobParameterDocumentId, new ActionListener<>() {
                     @Override
                     public void onResponse(AnomalyDetectorJob anomalyDetectorJob) {
-                        scheduledJobParameter = anomalyDetectorJob;
-                        inProgressFuture.complete(scheduledJobParameter);
+                        inProgressFuture.complete(anomalyDetectorJob);
```

Comment:
Don't we have to set `scheduledJobParameter = anomalyDetectorJob;` here?

Reply:
we set it below like this :

```
AnomalyDetectorJob scheduledJobParameter = inProgressFuture
                        .orTimeout(AnomalyDetectorSettings.REQUEST_TIMEOUT.get(settings).getMillis(), TimeUnit.MILLISECONDS)
                        .join();
```

Reply:
I get that. What I'm trying to ask is shouldn't it be inside `onResponse`?

Reply:
No, we only need to either `complete` or `completeExceptionally` within the `ActionListener`. Once `join` is invoked, that's where we can retrieve the response object

Reply:
This is an edit in a new file and not the old plugin code. Fine with this change!